internal/digraph/executor_validator.go

```diff
@@ -0,0 +1,73 @@
+package digraph
+
+import "fmt"
+
+// ExecutorValidator validates executor-specific configurations
+type ExecutorValidator interface {
```

Comment:
How about renaming it to `StepValidator`?
```suggestion
type StepValidator interface {
```

Reply:
I've renamed `ExecutorValidator` to `StepValidator` in `internal/digraph/executor/executor.go`. You're absolutely right - the interface validates steps, not executors themselves, so the name is now more accurate.

The interface is now:
`type StepValidator interface {
ValidateStep(step *digraph.Step) error
}`


Also, I've removed the old `internal/digraph/executor_validator.go` file as it's no longer needed with the refactored approach.